ios/engine/KMEI/KeymanEngine/Classes/Notification/Notifications.swift

```diff
@@ -41,10 +56,17 @@ public typealias LexicalModelRemovedNotification = InstallableLexicalModel
 public typealias LexicalModelPickerDismissedNotification = Void
 
 public struct Notifications {
+  public static let packageDownloadStarted = NotificationName<PackageDownloadStartedNotification>("KeymanPackageDownloadStarted")
+  public static let packageDownloadCompleted = NotificationName<PackageDownloadCompletedNotification>("KeymanPackageDownloadCompleted")
+  public static let packageDownloadFailed = NotificationName<PackageDownloadFailedNotification>("KeymanPackageDownloadFailed")
+
+  @available(swift, deprecated: 0.1, obsoleted: 0.1, message: "Download notifications are now package-based.  Use `packageDownloadStarted` instead.", renamed: "packageDownloadStarted")
```

Comment:
About this pattern:

When specifying `obsoleted`, any use of the affected property or method becomes a compile-time error.  **_Any_** use - even within other `obsoleted` methods or properties!  That's why the `typealias` definitions earlier in the file are merely `deprecated`.

Alas, Swift won't properly generate these compile errors with the `@available(*` syntax - a target (here, `swift`) must be specified.  Also, a Swift version number must be specified, even though we don't really care which version of Swift is used.  Ah well.